soccer/src/soccer/strategy/coach/coach_node.cpp

```diff
@@ -122,8 +134,109 @@ void CoachNode::check_for_play_state_change() {
 
         coach_state_pub_->publish(coach_message);
 
+        publish_static_obstacles();
+
         play_state_has_changed_ = false;
     }
 }
 
+void CoachNode::field_dimensions_callback(const rj_msgs::msg::FieldDimensions::SharedPtr& msg) {
+    current_field_dimensions_ = *msg;
+    have_field_dimensions_ = true;
+    publish_static_obstacles();
+}
+
+void CoachNode::publish_static_obstacles() {
+    if (have_field_dimensions_) {
+        // Publish defense areas as rectangular area obstacles
+        /* From the old gameplay node: "The defense area, per the rules, is the box
```

Comment:
you can delete "from the old gameplay node" here and just leave the explanation, as this won't make sense in like a year